tests and implementation of copying a dll resource to a tempFile

AhmetCanSolak opened the pull request:
provided method will be heavily used in projects which depend on `coremem`. Developed with in a TDD manner. You can find the tests and the implementation. Also updated `build.gradle` and `.gitignore`files accordingly. Please provide comment/review to merge.

royerloic commented on build.gradle:
```diff
@@ -85,6 +85,9 @@ dependencies
 	compile 'com.nativelibs4java:bridj:0.7.0'
 
+	// Commons:
+	compile ('commons-io:commons-io:2.4')                             {transitive= true}
```
Let's not get this huge dependency just for one method. Copy paste the function and adapt it for our needs instead...

AhmetCanSolak replied:
Okay. Indeed that one method was a really nice one but i will be coming with my best on this

royerloic commented on src/main/java/coremem/util/ResourceHandler.java:
```diff
@@ -0,0 +1,49 @@
+        )
+        {
+            byte[] buffer = new byte[4096];
+            for (;;) {
```
Are you sure you have to do this yourself? check online for less low-level approach that does not use dependencies..

royerloic commented on src/main/java/coremem/util/ResourceHandler.java:
```diff
@@ -0,0 +1,49 @@
+     * Method to copy a .dll resource from a jar
+     */
+    public File copyDLLfromJarToTempFile(String pRelativePathToFileInJar) throws IOException {
+        File lFile = File.createTempFile("coremem",".dll");
```
the file name should also depend on file name of the dll, just get pRelativePath... into a File and get the filename from it...

AhmetCanSolak replied:
That name is just for prefix. But still you are right i think, better to distinguish prefixes too. Will be updating that.

AhmetCanSolak commented:
Updated filename extraction part and added the loadResource directly functionality. One more commit will come for function naming conventions but you can provide review, next commit will not involve anything other than naming conventions.

royerloic commented:
travis ci is failing, once it passes we will merge

AhmetCanSolak commented:
> travis ci is failing, once it passes we will merge

https://travis-ci.org/ClearControl/coremem/jobs/487161424

Here it is. It is about `spotless.gradle` which is coming from master gradle repo.
